src/hotspot/share/services/management.cpp

```diff
@@ -2116,6 +2116,7 @@ JVM_ENTRY(jlong, jmm_GetTotalThreadAllocatedMemory(JNIEnv *env))
     }
 
     {
+      assert(MonitoringSupport_lock != nullptr, "Must be");
```

Comment:
This was unnecessary. If you have this here you would have one before every single use of mutex that is expected to be non-null.